internal/example/server.go

```diff
@@ -41,6 +43,58 @@ const (
 	grpcMaxConcurrentStreams = 1000000
 )
 
+type Server struct {
+	xdsserver server.Server
+}
+
+func NewServer(ctx context.Context, cache cache.Cache, cb *test.Callbacks) *Server {
+	srv := server.NewServer(ctx, cache, cb)
+	return &Server{srv}
+}
+
+func (s *Server) registerServer(grpcServer *grpc.Server) {
```

Comment:
Can we not just use the helper functions in the test package instead of duplicating that code?

https://github.com/envoyproxy/go-control-plane/blob/main/pkg/test/v3/register.go#L38

Same with `Run()` theres ` RunManagementServer` helper function.